src/connection.c

```diff
@@ -214,7 +208,11 @@ connection_new (gint       *client_fd,
         *client_fd = 0;
         return NULL;
     }
```

Comment:
GSocket is used to monitor G_IO_IN event, and this replaces the function select().
How about making use of the following functions to construct a GIOStream structure: g_unix_input_stream_new(), g_unix_output_stream_new() and g_simple_io_stream_new()? This GIOStream structure is used to read/write data from/to client and consistent with GTlsServerConnection/GTlsClientConnection.

Reply:
Do you know of a mechanism for getting G_IO_IN events from GInputStreams? I looked and was only able to find this mechanism in GSockets. Maybe I just didn't look hard enough.

Either way my goal is to do exactly what you suggest. I think that's the next step after this gets integrated and tested a bit.

Reply:
Ah I hadn't found the GPollableInputStream yet: https://developer.gnome.org/gio/stable/GPollableInputStream.html. Pretty sure this is what we need.

Reply:
Yes, agree.
With GPollableInputStream and GPollableOutputStream, it is feasible to handle read/write events in a stream level rather than the socket layer. The G_IO_IN events is no longer needed to generate an event source.